neuraxle/base.py

```diff
@@ -335,20 +337,39 @@ def __init__(
     def get_execution_mode(self) -> ExecutionMode:
         return self.execution_mode
 
-    def save_all_unsaved(self):
+    def save(self, full_dump=False):
         """
         Save all unsaved steps in the parents of the execution context using :func:`~neuraxle.base.BaseStep.save`.
         This method is called from a step checkpointer inside a :class:`Checkpoint`.
 
+        :param full_dump: save full pipeline dump to be able to load everything without source code (false by default).
+        :type full_dump: bool
         :return:
+
+        .. seealso::
+            :class:`BaseStep`,
+            :func:`~neuraxle.base.BaseStep.save`
         """
         while not self.empty():
             should_save_last_step = self.should_save_last_step()
             last_step = self.peek()
+            if full_dump:
+                should_save_last_step = True
 
             self.pop()
             if should_save_last_step:
-                last_step.save(self)
+                last_step.save(self, full_dump)
+
+    def save_last(self):
+        """
+        Save only the last step in the execution context.
```

Comment:
The concept of `last` is ambiguous here. Is it rather the innermost step? Whereas the outermost step would be the root pipeline. I would like to see a renaming or a more clear docstring. I'm not sure why we'd want to save just the last one, too. I have only quickly read the code, thought.

Reply:
also, if `save_last` and `should_save_last_step` are private, then it'd be good to add underscores to their names.